tck-test/TestFileParser.cpp

```diff
@@ -44,15 +41,28 @@ void TestFileParser::parseCommand(const std::string& command) {
     currentTest_.setName(parameters[1]);
     return;
   }
+  if (parameters.size() == 1 && (parameters[0] == "pass" || parameters[0] == "fail")) {
+    currentTest_.setSucceeding(parameters[0] == "pass");
+    return;
+  }
 
   TestCommand newCommand(std::move(parameters));
   if (!newCommand.valid()) {
-    LOG(ERROR) << "invalid command on line " << currentLine_ << ": " << command
-               << " (ignoring)";
+    LOG(ERROR) << "invalid command on line " << currentLine_ << ": " << command;
+    throw std::runtime_error("unknown commad in the test");
   } else {
     currentTest_.addCommand(std::move(newCommand));
   }
 }
 
+void TestFileParser::addCurrentTest() {
+  if (currentTest_.empty()) {
+    // no test defined
+  } else {
+    testSuite_.addTest(std::move(currentTest_));
+    DCHECK(currentTest_.empty());
```

Comment:
the std::move(currentTest_) on the line above is supposed to steal the guts from that instance and leave it in the empty state. The DCHECK is just a sanity check that it actually does that. If it works in debug build, it will work always.